### PR TITLE
306 build error

### PR DESCRIPTION
**Pull request by joeytalbot**
A couple of attempts at using 'if' or 'case_when' statements

### CODE REVIEW

File: _targets.R [hunk 1]
```diff
@@ -754,12 +754,29 @@ tar_target(zones_contextual, {
   zones <- sf::read_sf(file.path(tempdir(),"SIMD/simd2020_withgeog/sc_dz_11.shp"))
   simd <- read.csv(file.path(tempdir(),"SIMD/simd2020_withgeog/simd2020_withinds.csv"))
   
+  
+  simd = case_when(length(simd$`ï..Data_Zone`)>1 ~ {
+    simd = simd %>%
+      rename(Data_Zone = `ï..Data_Zone`)    
+  }, TRUE ~ simd)
```
Comment (Robinlovelace) [inline _targets.R:761]:
```suggestion
  if ("ï..Data_Zone" %in% names(simd)) {
    simd = simd %>%
      rename(Data_Zone = `ï..Data_Zone`)    
  }

File: _targets.R [hunk 2]
```diff
@@ -754,12 +754,29 @@ tar_target(zones_contextual, {
...
+  
+  # if(length(simd$`ï..Data_Zone`)>1) {
```
Comment (Robinlovelace) [inline _targets.R:763]: I think this comment should be tidied up.

File: _targets.R [hunk 3]
```diff
@@ -754,12 +754,29 @@ tar_target(zones_contextual, {
...
+  # }
+
+  
   unlink(file.path(tempdir(),"SIMD"), recursive = TRUE)
   
   zones <- zones[,c("DataZone","Name","TotPop2011","ResPop2011","HHCnt2011")]
   simd$Intermediate_Zone <- NULL
   simd$Council_area  <- NULL
   
+  # if(length(simd$`ï..Data_Zone`)>1) {
```
Comment (Robinlovelace) [inline _targets.R:775]: Is this supposed to be there?

File: code/tests/import_SMID.R [hunk 4]
```diff
@@ -10,7 +10,14 @@ files <- list.files(file.path(tempdir(),"SIMD/simd2020_withgeog"), full.names =
 
 zones <- read_sf(file.path(tempdir(),"SIMD/simd2020_withgeog/sc_dz_11.shp"))
 simd <- read.csv(file.path(tempdir(),"SIMD/simd2020_withgeog/simd2020_withinds.csv"))
-
+# if(length(simd$`ï..Data_Zone`)>1) {
+#   simd = simd %>% 
+#     rename(Data_Zone = `ï..Data_Zone`)    
+# }
+simd = case_when(length(simd$`ï..Data_Zone`)>1 ~ {
```
Comment (Robinlovelace) [inline code/tests/import_SMID.R:17]: Same as above: simple `if` statement is easier to read I think.

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
Few comments, then this should be good to go. Did you manage to get it working?

**Comment by juanfonsecaLS1** [comment]
you could try specifying the `encoding` parameter in the `read.csv` calls, if you don't want to change the function. 

[See this](https://stackoverflow.com/questions/14363085/invalid-multibyte-string-in-read-csv)

**Comment by Robinlovelace** [comment]
Fixed in #310

**Pull request closed without merging**